### DISCUSSION TITLE
Image export bug in Scratch

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#6568] [Ideas]
If I export the "ufo" costume (112x56) of the "ufo" sprite of project https://scratch.mit.edu/projects/669315467, I get a double size  "ufo.png" image (224x112). So, if I import it back in another project, the new costume is double the original size.

I would like that the width and height of exported bmp images are exactly the ones of the exported costume.

**Comment by BroJac5246** [comment]
I don't know for sure, but I would assume this is because Scratch uses quarter-pixels instead of full pixels. (Like how if you set the brush size to 1, the costume will only be 1x1 pixels if you make a 2x2 square.) Of course, by definition, pixels are as small as it gets, so these quarter-pixels are actually full pixels, despite what Scratch shows.

Because of that, I'm not sure if it makes sense to export at quarter-size because you'll lose detail. Instead, maybe importing could automatically resize the image to be quarter-size in Scratch. You're not actually losing quality, so it might be worth it.

The other option would be to fix this behavior altogether, but that would be complicated and cause lots of complications.

**Reply by Secret-chest** [reply]
I think the reasoning for this behaviour is fullscreen mode

**Comment by sfederici** [comment]
> maybe importing could automatically resize the image to be quarter-size in Scratch.

Would that impact only images previously exported from Scratch? Of course we want a costume created importing an image being the same size as the image, right?

**Reply by BroJac5246** [reply]
> Would that impact only images previously exported from Scratch?

Probably not because that would be very hard to determine (unless we add it to the metadata upon an export?).

> Of course we want a costume created importing an image being the same size as the image, right?

It would, of course, be ideal if the size of the image is consistent with the one Scratch has, so that's why I'm proposing quarter-size importing. You won't be losing detail and the image will be the same size.

**Comment by sfederici** [comment]
> It would, of course, be ideal if the size of the image is consistent with the one Scratch has, so that's why I'm proposing quarter-size importing. You won't be losing detail and the image will be the same size.

I understand that. And that would solve my problem. But I think that it would be confusing if, e.g., a 100x100 image is imported and the costume size is 50x50.

**Reply by BroJac5246** [reply]
But do you generally take note of the dimensions of the image before you upload it? I would guess not.

Do you have any ideas on how this could be solved, potentially based on the presented ideas?

**Reply by sfederici** [reply]
I usually don't take note of the dimensions, but sometimes I do, so I would be suprised if the image dimensions are not the ones I know.

My idea would be that exported costumes should have the same dimensions of the original image, even if this has an impact on the image definition.

**Reply by BroJac5246** [reply]
> I usually don't take note of the dimensions, but sometimes I do, so I would be suprised if the image dimensions are not the ones I know.
> 
> My idea would be that exported costumes should have the same dimensions of the original image, even if this has an impact on the image definition.

I'm not sure if this is possible without major changes to how Scratch works, though. That would mean Scratch Addons would have to fix everything in the background and that would get complicated fast.

I think the best and only realistic solutions are either automatically resizing everything or adding metadata and resizing based on that.

**Comment by sfederici** [comment]
It the metadata solutions is quickly feasible, I guess it would work better. If instead this is too long to be implemented, then having an automatic resize when loading images could work for me and my students. So, at least they will be able to export images from a project and then import them to anothee project without having to understand why the imported image is double the size of the original one.
